### PR TITLE
Define the administrator role and its rules.

### PR DESCRIPTION
**Pull request by artursouza**
Signed-off-by: Artur Souza <artursouza.ms@outlook.com>

### CODE REVIEW

File: ADMINISTRATORS.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Dapr administrators
+
+## Charter
+
+Administrators have access to credentials equivalent to members of the steering and technical commitee (STC) and are responsible to perform pre-approved routine maintenance duties and ad-hoc tasks delegated by the STC without making isolated decisions. This document outlines a list of pre-approved duties and some that require STC approval - when in doubt, the administrator should seek guidance from one STC member.
+
+### Pre-approved duties
+
+An administrator might perform the following duties without requiring approval from any STC member:
```
Comment (msfussell) [inline ADMINISTRATORS.md:9]: Given these duties, this seems like there needs to be a new administrators group in the github that has these privileges. Similar to the release team account. If so, can you add into the steps below that on approval by STC, administrators are added to this account? This is in addition to having access to the STC vault in 1Password for credentials.

File: ADMINISTRATORS.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+* Rotate credentials to services currently enrolled
+* Update credentials on GitHub
+* Rerun failed test runs
+* Provision cloud resources to existing enrolled cloud providers
+* Fix operational issues with existing services the project depends on (GitHub Actions, FOSSA, etc)
+* Force merge a pull request upon request by repository's maintainer
+
+### Duties that require STC approval
+
+* Add or remove approvers, maintainers or administrators
```
[the remark this answers is not in the document]
Comment (artursouza) [reply ADMINISTRATORS.md:20]: I thought that an admin could also perform the duties to add a new approver or maintainer AFTER STC approval. Any reason not to?

File: ADMINISTRATORS.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+* STC votes with 3/5 majority to remove administrator
+* Administrator joins as an STC member
+
+## Suspension
+
+In order to de-risk potential misuse of power or suspicion of leaked credentials, at any moment, any STC member can temporarily suspend administrator role from any individual - revoking all prioviledge access - for any reason, without requiring an STC meeting. Reinstating administrator role requires the same process as becoming one in the first place.
+
+## Current Administrators
+
+| Name | Handle | Company | Status | Timezone | Term Start | Term End |
```
[the remark this answers is not in the document]
Comment (artursouza) [reply ADMINISTRATORS.md:54]: For people to know who is available at their TZ.

File: ADMINISTRATORS.md (hunk 1 above)
Comment (artursouza) [reply ADMINISTRATORS.md:9]: I agree.

File: ADMINISTRATORS.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## Process to become an administrator
+
+0. As a pre-requisite, an administrator candidate **must first become a maintainer** of at least one repository in [Dapr](https://github.com/dapr). Being a maintainer in [dapr-sandbox](https://github.com/dapr-sandbox) does not qualify.
+
+1. Then, the candidate can self-nominate via a [new issue in the Community repository](https://github.com/dapr/community/issues/new), stating which repositories candidate currently maintains.
+
+2. In addition, candidate should add a link to the new issue asking it to be voted in the next STC meeting. Agenda for STC meetings are discussed in [issues in the Community repository](https://github.com/dapr/community/issues). Alternatively, candidate can request one STC member to seek approval via e-mail instead of waiting for the next STC meeting.
+
+3. Finally, a maintainer will become an administrator after STC approves the request with a 3/5 majority.
```
[the remark this answers is not in the document]
Comment (artursouza) [reply ADMINISTRATORS.md:32]: It is important to determine with type of majority. Simple, 2/3 or 3/5?

File: ADMINISTRATORS.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+The administrator role expires 6 months from the date it was approved. The process for renewal is the same as to become an administrator in the first place.
+
+To minimize the number of people with access to shared credentials, there can only be 3 active or suspended administrators. Adding one more requires the STC to remove another.
+
+## Removal
+
+The administrator role is permanently removed in case any of the following:
+* Expiration of the 6 months term as administrator
+* Administrator is no longer a maintainer in Dapr
+* STC votes with 3/5 majority to remove administrator
```
[the remark this answers is not in the document]
Comment (artursouza) [reply ADMINISTRATORS.md:45]: We need to specify which type of majority.

File: ADMINISTRATORS.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (msfussell) [reply ADMINISTRATORS.md:20]: I read this wrong. I thought this means STC adding. You can ignore this comment.

File: ADMINISTRATORS.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (yaron2) [reply ADMINISTRATORS.md:32]: Let's use simple

File: ADMINISTRATORS.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (yaron2) [reply ADMINISTRATORS.md:45]: Agree, need to specify. Let's use simple majority

File: ADMINISTRATORS.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (msfussell) [reply ADMINISTRATORS.md:32]: Ok. Just feel like this is more complex, given that 3/5 of 7 is 4.2, so is that 4 people out of 7? And 3/5 of 9 is 5.4 so is that 5 people? 3/5 of 11 of 6.6 so this is 7 people. Then this is just a majority in all cases.

File: ADMINISTRATORS.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (msfussell) [reply ADMINISTRATORS.md:45]: Same as above.

File: ADMINISTRATORS.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (msfussell) [reply ADMINISTRATORS.md:54]: Ok

File: ADMINISTRATORS.md (hunk 1 above)
Comment (msfussell) [reply ADMINISTRATORS.md:9]: I didn't see any update below on this comment. I suggest creating a github team called `administrators` here https://github.com/orgs/dapr/teams and then saying that this is what makes you and admin as well as access to the STC 1password vault.

Comment (artursouza) [reply ADMINISTRATORS.md:9]: I am adding the text now. GitHub does not allow a Team to be made org owner, only individuals. So, we cannot really create an `administrators` role.

File: ADMINISTRATORS.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (artursouza) [reply ADMINISTRATORS.md:32]: Done.

File: ADMINISTRATORS.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (artursouza) [reply ADMINISTRATORS.md:45]: Doing simple.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**